rust/kernel/file_operations.rs

```diff
@@ -100,8 +100,7 @@ unsafe extern "C" fn open_callback<A: FileOpenAdapter, T: FileOpener<A::Arg>>(
         // be a valid non-null pointer.
         let ptr = T::open(unsafe { &*arg })?.into_pointer();
         // SAFETY: `ptr` was previously returned by `T::open`. The returned
-        // value should be a boxed value and should live the length of the
-        // given file.
+        // value should be a boxed value and should outlive the given file.
```

Comment:
The original justification was just wrong here.

Note that we don't need to say anything about `ptr`, we just need to say that the caller guarantees that `file` is writable and aligned so that we're ok writing to its `private_data` field.